client/components/Dashboard/VersionChecker.js

```diff
@@ -9,31 +9,37 @@ class VersionChecker {
     checkForNewVersion(currentVersion) {
         this.currentVersion = currentVersion;
 
-        console.log('Checking for a new server version...');
-        console.log(`Dashboard running on version ${currentVersion}`);
-
         this.xmlHttp = new XMLHttpRequest();
         this.xmlHttp.onreadystatechange = () => this.onVersionResult();
         this.xmlHttp.open('GET', '/version', true);
         this.xmlHttp.send(null);
     }
 
     onVersionResult() {
-        if (this.xmlHttp.readyState == 4 && this.xmlHttp.status == 200) {
-            const result = JSON.parse(this.xmlHttp.responseText);
-            console.log(`Server running on version ${result.version}`);
-
-            if (semver.lt(this.currentVersion, result.version)) {
-                this.refreshDashboard();
-                return;
-            }
-            console.log(`Dashboard still up-to-date!`);
+        if (this.xmlHttp.readyState !== 4) {
+            return;
+        }
+
+        if (this.xmlHttp.status !== 200) {
+            console.log(`Failed to fetch the CIMonitor server version.`);
+            return;
+        }
+
+        const result = JSON.parse(this.xmlHttp.responseText);
+
+        if (semver.lt(this.currentVersion, result.version)) {
+            this.refreshDashboard(result.version);
+            return;
         }
+
+        console.log(`Dashboard version ${this.version} up-to-date with server version ${result.version}.`);
     }
 
-    refreshDashboard() {
-        console.log('Refreshing in 10 seconds to be up-to-date');
+    refreshDashboard(serverVersion) {
```

Comment:
I would call this showUpdateNotificationOnDashboard :)

Reply:
Here it actually refreshes the dashboard as it's the client side code :innocent: